framework/src/meshgenerators/FancyExtruderGenerator.C

```diff
@@ -135,6 +177,31 @@ FancyExtruderGenerator::generate()
   auto mesh = buildMeshBaseObject();
   mesh->set_mesh_dimension(_input->mesh_dimension() + 1);
 
+  // Check if the element integer names are existent in the input mesh.
+  for (unsigned int i = 0; i < _elem_integer_names_to_swap.size(); i++)
+    if (_input->has_elem_integer(_elem_integer_names_to_swap[i]))
+      _elem_integer_indices_to_swap.push_back(
+          _input->get_elem_integer_index(_elem_integer_names_to_swap[i]));
+    else
+      paramError("elem_integer_names_to_swap",
+                 "Element ",
+                 i + 1,
+                 " of elem_integer_names_to_swap in ",
+                 name(),
+                 " is not a valid extra element integer of the input mesh.");
+
+  // prepare for transferring extra element integers from orignal mesh to the extruded mesh.
+  const unsigned int num_extra_elem_integers = _input->n_elem_integers();
+  std::vector<std::string> id_names;
+
+  for (unsigned int i = 0; i < num_extra_elem_integers; i++)
+    id_names.push_back(_input->get_elem_integer_name(i));
+
+  // retreive subdomain/sideset/nodeset name maps
+  auto input_subdomain_map = _input->get_subdomain_name_map();
+  auto input_sideset_map = _input->boundary_info->get_sideset_name_map();
+  auto input_nodeset_map = _input->boundary_info->get_nodeset_name_map();
```

Comment:
```suggestion
  const auto & input_subdomain_map = _input->get_subdomain_name_map();
  const auto & input_sideset_map = _input->get_boundary_info()->get_sideset_name_map();
  const auto & input_nodeset_map = _input->get_boundary_info()->get_nodeset_name_map();
```

We try to use the APIs (here, for boundary_info) as much as possible. The fact that boundary info is a public member variable is just something we're stuck with in libmesh due to deprecation